src/gitgraph.js

```diff
@@ -776,6 +816,37 @@
     }
   };
 
+  /**
+   * Push a new point to path.
+   * This method will combine duplicate points and reject reversed points.
+   * 
+   * @this Branch
+   */
+  Branch.prototype.pushPath = function (point) {
+    var lastPoint = this.path.slice( -1 )[ 0 ];
+    if ( !lastPoint ) {
+      this.path.push( point );
+    } else if ( lastPoint.x === point.x && lastPoint.y === point.y ) {
+      if ( lastPoint.type !== "start" && point.type === "end" ) {
+        lastPoint.type = "end";
+      } else if ( point.type === "join" ) {
```

Comment:
I guess this `else if` / `else` could be refactor into one `else if ( point.type !== "join" )`